semantic-python/src/Language/Python/Core.hs

```diff
@@ -218,17 +218,17 @@ instance Compile Py.ExpressionStatement where
   compile it@Py.ExpressionStatement
     { Py.extraChildren = children
     } cc = do
-    foldr compile cc children >>= locate it
+    foldr compile cc children >=> locate it
 
 instance Compile Py.ExpressionList where
   compile it@Py.ExpressionList { Py.extraChildren = [child] } cc
-    = compile child cc >>= locate it
+    = compile child cc >=> locate it
   compile Py.ExpressionList { Py.extraChildren = items } _
-    = fail ("unimplemented: ExpressionList of length " <> show items)
+    = const (fail ("unimplemented: ExpressionList of length " <> show items))
 
 
 instance Compile Py.False where
-  compile it _ = locate it $ bool False
+  compile it cc _ = locate it (bool True) >>= cc
```

Comment:
Note that we no longer ignore the continuation here, because we have the ability for a caller to specify that the continuation can be just `pure`.